Cargo.toml

```diff
@@ -20,3 +20,15 @@ version = "0.2.2"
 bitflags = "1.2.1"
 embedded-hal = "0.2.4"
 nb = "0.1.2"
+
+# dependencies used in examples, dev-dependencies are not allowed to be optional
+cortex-m = { version = "0.7.2", optional = true }
+cortex-m-rt = { version = "0.6.13", optional = true }
+heapless = { version = "0.7.0", optional = true }
+panic-rtt-target = { version = "0.1.2", features = ["cortex-m"], optional = true }
+rtt-target = { version = "0.3.1", features = ["cortex-m"], optional = true }
+stm32f1xx-hal = {version = "0.7.0", features = ["stm32f103", "rt"], optional = true }
```

Comment:
I think you can just put these in `[dev-dependencies]` and just have them as not optional, no?

Reply:
That was my original intention, but `[dev-dependencies]` cannot be optional (I didn't know that either - see https://github.com/rust-lang/cargo/issues/1596 for endless discussion and referenced PRs from vairous projects :-))

Reply:
Oh, I see, to quick on reading. So having them as not optional would mean they're always compiled and that would cause the tests not to run - as these need to run in `std` env on a std target (it runs on `x86_64-unknown-linux-gnu` target in the CI)

Reply:
Oh ok, I see the dilemma then. Maybe instead of the `examples/` folder (and using cargo's support for examples) we just put the bluepill example code in a new Cargo project (something like `bluepill-example`). Its a little weird, but that gets around the dev-dependencies and having the dependencies as optional

Reply:
That is a great idea - I've moved the examples to separate crate `rn4870-examples` marked as `publish = false`.